### ISSUE TITLE
Placing book in Lectern fires block place event

### ISSUE THREAD

**Issue by tal5** [PaperSpigot/Paper#9759]
### Expected behavior

The JD for the block place event says `Called when a block is placed by a player.`, so either for the event to not fire there or for the JD to be clarified.

### Observed/Actual behavior

Whenever you place a book into a Lectern, a block place event is fired.

### Steps/models to reproduce

This is using the Debuggery plugin (which is part of Paper's org), let me know if proper Java code is preferred.

- `/devent org.bukkit.event.block.BlockPlaceEvent`
- Right click an empty Lectern with a book

### Plugin and Datapack List

```
> plugins
[23:44:43 INFO]: Server Plugins (1):
[23:44:43 INFO]: Bukkit Plugins:
[23:44:43 INFO]:  - Debuggery
```
```
> datapack list
[23:46:10 INFO]: There are 2 data pack(s) enabled: [vanilla (built-in)], [file/bukkit (world)]
[23:46:10 INFO]: There are no more data packs available
```

### Paper version

```
> version
[23:49:40 INFO]: Checking version, please wait...
[23:49:41 INFO]: This server is running Paper version git-Paper-196 (MC: 1.20.1) (Implementing API version 1.20.1-R0.1-SNAPSHOT) (Git: 773dd72)
You are running the latest version
Previous version: git-Paper-194 (MC: 1.20.1)
```

### Other

Some [Discord discussion](https://discord.com/channels/289587909051416579/555462289851940864/1155629979032371211) if that's relevant.

**Comment by Brokkonaut** [comment]
I would prefer to have the javadocs changed to reflect that behavior instead of breaking plugins...

**Comment by Owen1212055** [comment]
I disagree, this really does not seem like intended behavior. But at the same time, this event's current implementation makes stuff like that very confusing.

**Comment by Brokkonaut** [comment]
even if it was not intended, it is the only useful event atm (also for waxing, unwaxing, log stripping,.. - playerinteract does not have the resulting change result so it is quite bad to use for this), so it would break nearly every protection plugin if it is changed.

and if it is documented it can be made intended behavior.

**Comment by Leguan16** [comment]
100% agree with brokkonaut here.

**Comment by Leguan16** [comment]
I have looked into this for a while and discovered that we can just create a `CraftItemStack` out of the used item and check if it is an actual block. That should fix the issue with lecterns and copper blocks. 
But what are we doing now?

**Comment by Warriorrrr** [comment]
Stuff like this shouldn't really be intended behavior as it makes no sense, new API should be added instead. In this case there's already a PR open that adds an insert book event

**Comment by Leguan16** [comment]
There were never talks about whether it should be considered intended behavior or not. 

Talks currently are at that point that several people said that the current `BlockPlaceEvent` should be declared as deprecated due to its inconsistency and to introduce [a new event](https://canary.discord.com/channels/289587909051416579/555462289851940864/1156293645142605936).

**Comment by Djaytan** [comment]
I just filled this issue: https://github.com/PaperMC/Paper/issues/12256
Then I discovered this one is related to what I wanted to report.

Someone reported me a bug with my plugin and, after investigations, I discovered it was due to weird behaviors from the API (more details [here](https://www.spigotmc.org/threads/jobsreborn-patchplacebreak-addon.561392/page-5#post-4835329)).

I don't know if there is any progress so far related to this topic, the eventual depreciation of `BlockPlaceEvent` and so on...
